lib/absinthe/schema/notation.ex

```diff
@@ -34,15 +34,45 @@ defmodule Absinthe.Schema.Notation do
   end
 
   @doc """
-  Set the topic function for a subscription field. See the `subscription/2` macro
-  docs for details
+  REPLACED BY
+
+  Remove in final 1.4.0 release
   """
-  defmacro topic(topic_fun) do
-    __CALLER__.module
-    |> Scope.put_attribute(:topic, topic_fun)
-    :ok
+  defmacro topic(_) do
+    raise "`topic/1` macro is no longer used! replaced by `config/1`"
+  end
+
+  @doc false
+  defmacro resolver(_) do
+    raise "`resolver/1` is not a function, did you mean `resolve` ?"
+  end
+
+  @placement {:config, [under: [:field]]}
+  @doc """
+  Configure a subscription field.
+
+  ## Example
+
+  ```elixir
+  config fn args, %{context: context} ->
+    if authorized?(context) do
+      {:ok, topic: args.client_id}
+    else
+      {:error, "unauthorized"}
+    end
+  end
```

Comment:
@freshtonic ^

Note that we're doing away with the `topic` macro entirely in favor of this `config` macro. It gives us the possibility of setting up more than just the topic, which is valuable for a variety of other features we're looking at.

Reply:
I like this!